assets/js/blocks-registry/payment-methods/assertions.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { isValidElement } from '@wordpress/element';
+
+export const assertValidPaymentMethodComponent = (
+	component,
+	componentName
+) => {
+	// @todo detect if functional component (not render prop)
+	if ( typeof component !== 'function' ) {
+		throw new TypeError(
+			`The ${ componentName } for the payment method must be a functional component`
+		);
+	}
+};
+
+export const assertValidElement = ( element, elementName ) => {
+	if ( ! isValidElement( element ) && element !== null ) {
```

Comment:
Considering `element !== null` is less expensive than the first check, what do you think about switching the order of the two?
```suggestion
	if ( element !== null && ! isValidElement( element ) ) {
```

Reply:
Agreed. Or maybe just putting a `null` check as the first check inside `isValidElement`?

Reply:
`isValidElement` is actually `React.isValidElement` so we can't modify that unless we create a wrapper.  I agree though that having the less expensive check first is preferable and will do.

Reply:
Done in c833a83